File: java/Student.java
import java.util.Scanner;

public class Student {
    private String nama;
    private String npm;
    private Nilai penguji1;
    private Nilai penguji2;
    private Nilai pembimbing;

    Student() {
        this.nama="";
        this.npm="";
        this.penguji1=new Nilai();
        this.penguji2=new Nilai();
        this.pembimbing=new Nilai();
    }

    Student(String nama, String npm, Nilai penguji1, Nilai penguji2, Nilai pembimbing) {
        this.nama=nama;
        this.npm=npm;
        this.penguji1=penguji1;
        this.penguji2=penguji2;
        this.pembimbing=pembimbing;
    }

    public void input() {
        Scanner input=new Scanner(System.in);

        System.out.print("Masukkan nama\t\t\t: ");
        this.nama=input.nextLine();

        System.out.print("Masukkan npm\t\t\t: ");
        this.npm=input.nextLine();

        penguji1.input("Penguji 1");
        penguji2.input("Penguji 2");
        pembimbing.input("Pembimbing");
    }

    public void print() {
        System.out.print("| " + this.npm + "\t");
        System.out.print("| " + this.nama + " \t\t");
        System.out.print("| " + this.penguji1.getNilai() + "\t\t");
        System.out.print("| " + this.penguji2.getNilai() + "\t\t");
        System.out.print("| " + this.pembimbing.getNilai() + "\t\t");
        System.out.print("| " + this.getNilaiSidang().getNilai() + "\t\t");
        System.out.println("| " + this.getHurufMutu(this.getNilaiSidang().getNilai()) + "\t\t|");
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setNpm(String npm) {
        this.npm = npm;
    }

    public void setPenguji1(Nilai penguji1) {
        this.penguji1 = penguji1;
    }

    public void setPenguji2(Nilai penguji2) {
        this.penguji2 = penguji2;
    }
    
    public void setPembimbing(Nilai pembimbing) {
        this.pembimbing = pembimbing;
    }

    public String getNama() {
        return nama;
    }

    public String getNpm() {
        return npm;
    }

    public Nilai getPenguji1() {
        return penguji1;
    }

    public Nilai getPenguji2() {
        return penguji2;
    }

    public Nilai getPembimbing() {
        return pembimbing;
    }

    public Nilai getNilaiSidang() {
        float nilai1=30*this.penguji1.getNilai()/100;
        float nilai2=30*this.penguji2.getNilai()/100;
        float nilai3=40*this.pembimbing.getNilai()/100;
        Nilai sidang=new Nilai(nilai1+nilai2+nilai3);
        return sidang;
    }

    public char getHurufMutu(float nilai) {
        if(nilai > 80 && nilai <= 100) {
            return 'A';
        } else if(nilai > 70 && nilai <= 80) {
            return 'B';
        } else if(nilai > 60 && nilai <= 70) {
            return 'C';
        }
        return 'T';
    }
}

File: py/Student.py
from Nilai import Nilai

class Student :
    def __init__(self) :
        self.__nama=""
        self.__npm=""
        self.__penguji1=Nilai()
        self.__penguji2=Nilai()
        self.__pembimbing=Nilai()

    def input(self) :
        self.__nama=input("Masukkan nama: ")
        self.__npm=input("Masukkan NPM: ")
        self.__penguji1.input("Penguji 1")
        self.__penguji2.input("Penguji 2")
        self.__pembimbing.input("Pembimbing")

    def print(self) :
        print("| " + self.__npm + "\t| " 
             + self.__nama + " \t\t| " 
             + str(self.__penguji1.getNilai()) + "\t\t| " 
             + str(self.__penguji2.getNilai()) + "\t\t| " 
             + str(self.__pembimbing.getNilai()) + "\t\t| " 
             + str(self.getNilaiSidang().getNilai()) + "\t\t| " 
             + self.getHurufMutu() + "\t\t|"
        )

    def setNama(self,nama) : self.__nama=nama
    def setNpm(self,npm) : self.__npm=npm
    def setPenguji1(self,penguji1) : self.__penguji1=penguji1
    def setPenguji2(self,penguji2) : self.__penguji2=penguji2
    def setPembimbing(self, pembimbing) : self.__pembimbing=pembimbing

    def getNama(self) : return self.__nama
    def getNpm(self) : return self.__npm
    def getPenguji1(self) : return self.__penguji1
    def getPenguji2(self) : return self.__penguji2
    def getPembimbing(self) : return self.__pembimbing

    def getNilaiSidang(self) :
        nilai1=self.__penguji1.getNilai()
        nilai2=self.__penguji2.getNilai()
        nilai3=self.__pembimbing.getNilai()
        sidang=Nilai()
        sidang.setNilai(nilai1+nilai2+nilai3)
        return sidang

    def getHurufMutu(self,nilai) :
        if nilai > 80 and nilai <= 100 :
            return 'A'
        elif nilai > 70 and nilai <= 80 :
            return 'B'
        elif nilai > 60 and nilai <= 70 :
            return 'C'
        return 'T'